extensions/wikia/ArticleHeader/PageTitle.class.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace ArticleHeader;
+
+class PageTitle {
+
+    /* @var string */
+    private $title;
+
+    /* @var \WikiaGlobalRegistry */
+    private $wg;
+
+    /* @var int */
+    private $namespace;
+
+    /**
+     * @param \WikiaApp $app
+     */
+    public function __construct($app) {
+        $this->wg = $app->wg;
+
+        $this->title = $app->getSkinTemplateObj()->data['title'];
+        $this->namespace = $this->wg->Title->getNamespace();
+    }
+
+    /**
+     * @return string
+     */
+    private function titleSpecial() {
+        return $this->title;
+    }
+
+    /**
+     * @return string
+     */
+    private function titleArticle() {
+        // remove namespaces prefix from title
+        $namespaces = [ NS_MEDIAWIKI, NS_TEMPLATE, NS_CATEGORY, NS_FILE ];
+
+        if (
+            in_array( $this->namespace,
+                array_merge( $namespaces, $this->wg->SuppressNamespacePrefix )
+            )
+        ) {
+            $title = $this->wg->Title->getText();
+        }
+
+        return $this->title;
+    }
+
+    /**
+     * @return string
+     */
+    private function titleForum() {
+        return $this->wg->Title->getText();
+    }
+
+    /**
+     * @return string
+     */
+    private function titleUserPages() {
+        $title = explode( ':', $this->title, 2 ); // User:Foo/World_Of_Warcraft:_Residers_in_Shadows (BAC-494)
+        if ( count( $title ) >= 2 && $this->wg->Title->getNsText() == str_replace( ' ', '_', $title[0] ) ) // in case of error page (showErrorPage) $title is just a string (cannot explode it)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)